test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,209 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+            cql.execute(insert, [0, 0, 10])
+            cql.execute(insert, [0, 1, 20])
+
+            # Delete one of the partitions
+            cql.execute(f"DELETE FROM {table} WHERE p1=0 AND p2=0")
+
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=0"))
+            assert [20] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=1")]
+
+            assert [(1,20)] == [(x.p2, x.c) for x in cql.execute(f"SELECT p2, c FROM {mv} WHERE p1=0")]
+
+# Test deleting a large partition when there is a view with the same partition
+# key, and verify that view updates metrics is increased by exactly 1. Deleting
+# a partition in this case is expected to generate one view update for deleting
+# the corresponding view partition by a partition tombstone.
+def test_base_partition_deletion_with_same_view_partition_key_metrics(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key (p,c)') as table:
+        # Insert into two base partitions. We will delete one of them
+        v = 42
+        insert = cql.prepare(f'INSERT INTO {table} (p,c,v) VALUES (?,?,{v})')
+        # Create a view with the same partition key as the base.
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null') as mv:
+            N = 1000
+            for i in range(N):
+                cql.execute(insert, [1, i])
+
+            # Before the deletion, all N rows should exist in the base and the view
+            allN = list(range(N))
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=1")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv} WHERE p=1")])
+
+            metrics_before = ScyllaMetrics.query(cql)
+            updates_before = metrics_before.get('scylla_database_total_view_updates_pushed_local')
+
+            cql.execute(f"DELETE FROM {table} WHERE p=1")
+
+            # After the deletion, all data should be gone from both base and view
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv} WHERE p=1"))
+
+            metrics_after = ScyllaMetrics.query(cql)
+            updates_after = metrics_after.get('scylla_database_total_view_updates_pushed_local')
+
+            print(f"scylla_database_total_view_updates_pushed_local: {updates_before} -> {updates_after}")
+            assert updates_after == updates_before + 1
+
+# Same as above test, except the view partition key is a permutation of
+# the base partition key. The delete partition optimization should still
+# apply and delete the view partition in one update.
+def test_base_partition_deletion_with_permutated_view_partition_key_metrics(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key ((p,c),v)') as table:
```

Comment:
If (p,c) are two partition key components, please consider naming them p1,p2 instead of p,c. Otherwise it sounds like c is - as in other tests - a clustering key.